networkx/algorithms/dag.py

```diff
@@ -788,6 +884,12 @@ def dag_longest_path(G, weight="weight", default_weight=1, topo_order=None):
     NetworkXNotImplemented
         If `G` is not directed
 
+    Examples
+    --------
+    >>> DG = nx.DiGraph([(1, 2), (2, 3)])
+    >>> nx.dag_longest_path(DG)
+    [1, 2, 3]
```

Comment:
It might be nice to have an example where there are multiple paths to the same node to better illustrate the "longest path" idea, e.g.

```suggestion
    >>> DG = nx.DiGraph([(0, 1), (1, 2), (0, 2)])
    >>> nx.all_simple_paths(DG, 0, 2)
    [[0, 1, 2], [0, 2]]
    >>> nx.dag_longest_path(DG)
    [0, 1, 2]
```